src/uct/ib/base/ib_iface.c

```diff
@@ -930,11 +937,18 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
     }
 
     status = uct_ib_device_query_gid(dev, self->config.port_num,
-                                     self->config.gid_index, &self->gid);
+                                     self->config.gid_index, &self->gid,
+                                     &is_roce_v2);
     if (status != UCS_OK) {
         goto err;
     }
 
+    if (is_roce_v2 && (config->traffic_class == UCS_CONFIG_ULUNITS_AUTO)) {
+        self->config.traffic_class = UCT_IB_DEFAULT_ROCEV2_DSCP;
+    } else {
+        self->config.traffic_class = config->traffic_class;
```

Comment:
what if `is_roce_v2` is `false`, but `traffic_class` is `UCS_CONFIG_ULUNITS_AUTO`?